File: GlobalHub/Services/ProjectService/ProjectService.Application/Dto/ProjectDto.cs
namespace ProjectService.Application.Dto;

public record ProjectDto
{
    public long Id { get; init; }

    public string Title { get; init; }

    public ICollection<TagDto> Tags { get; init; }

    public ICollection<ProjectItemDto> ProjectItems { get; init; }

    public DateTimeOffset CreatedDate { get; init; }

    public DateTimeOffset? UpdatedDate { get; init; }

    public string CreatedBy { get; init; }

    public string? UpdatedBy { get; init; }
}

File: ClientApp/src/dto/projects/projectDto.ts
import { IHasDate } from '../../interfaces/IHasDate';
import { IHasAuthor } from '../../interfaces/IHasAuthor';
import { ProjectTagDto } from './tags/projectTagDto';
import { ProjectItemDto } from './projectItemDto';

export interface ProjectDto extends IHasDate, IHasAuthor {
  id: number;
  title: string;
  tags?: ProjectTagDto[];
  projectItems?: ProjectItemDto[];
}
